### ISSUE TITLE
Uncouple flymake from eglot

### ISSUE THREAD

**Issue by jvillasante** [joaotavora/eglot#123]
Is there a way to switch to flycheck instead of flymake?

**Comment by joaotavora** [comment]
No, you have to write a flycheck backend first. The easiest way would be to
write a flycheck backend for flymake.

On Sep 30, 2018 21:45, "Julio C. Villasante Gomez" <notifications@github.com>
wrote:

Is there a way to switch to flycheck instead of flymake?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/joaotavora/eglot/issues/123>, or mute the thread

**Comment by jvillasante** [comment]
Ok, awesome!
Is there a way to disable flymake?

**Comment by joaotavora** [comment]
M-x flymake-mode, but you'd probably break other parts of eglot in the
pieces. You can open an issue to make eglot less dependent on flymake.

**Comment by TatriX** [comment]
Can I somehow help with the flycheck integration?

**Comment by joaotavora** [comment]
@TatriX I'm not going to integrate Flycheck into eglot.  Eglot is always going to use Flymake's API.  But you can make a Flycheck to Flymake adapter, just read the Flymake manual in Emacs, `M-x info` then search for "flymake", then "Extending flymake" then "Backend functions".  Understand how a backend function works and write a Flycheck checker that properly calls it.  The docstring of `flymake-diagnostic-functions` is also useful for that.

**Comment by q3cpma** [comment]
I'm also interested in disabling flymake, since I only want and expect completion and variable name change from LSP.

**Comment by joaotavora** [comment]
This is silly... :blush: I had already fixed this back in May in c8e7ab0c6e1cbd4902fdac6ce0435e84f9790b25 . 

So there shouldn't really be a problem if you disable flymake in Eglot-managed buffers.

The way to do it automatically isn't particularly pretty, but it should work fine:

```
(add-hook 'eglot--managed-mode-hook (lambda () (flymake-mode -1)))
```
